test/src/unittests/tonal/test_pitchcontours.py

```diff
@@ -0,0 +1,246 @@
+        #    spectrum -> spectral peaks -> pitch salience function -> pitch salience function peaks
+        for frame in FrameGenerator(audio, frameSize=frameSize, hopSize=hopSize):
+            frame = run_windowing(frame)
+            spectrum = run_spectrum(frame)
+            peak_frequencies, peak_magnitudes = run_spectral_peaks(spectrum)
+            
+            salience = run_pitch_salience_function(peak_frequencies, peak_magnitudes)
+            salience_peaks_bins, salience_peaks_saliences = run_pitch_salience_function_peaks(salience)
+            
+            pool.add('allframes_salience_peaks_bins', salience_peaks_bins)
+            pool.add('allframes_salience_peaks_saliences', salience_peaks_saliences)
+
+        # 3. Now, as we have gathered the required per-frame data, we can feed it to the contour 
+        #    tracking and melody detection algorithms:
+        contours_bins, contours_saliences, contours_start_times, duration = run_pitch_contours(
+                pool['allframes_salience_peaks_bins'],
+                pool['allframes_salience_peaks_saliences'])
+
+
+        save('calculatedcontour_bins_scale0.npy', contours_bins[0])
+        save('calculatedcontour_saliences_scale0.npy', contours_saliences[0])
+        save('calculatedcontour_bins_scale1.npy', contours_bins[1])
+        save('calculatedcontour_saliences_scale1.npy', contours_saliences[1])
+        save('calculatedcontour_startTimes_scale.npy', contours_start_times)
+
+        expectedBins0 = load(join(filedir(), 'pitchsalience/calculatedcontour_bins_scale0.npy'))
+        expectedSaliences0 = load(join(filedir(), 'pitchsalience/calculatedcontour_saliences_scale0.npy'))
+        expectedBins1 = load(join(filedir(), 'pitchsalience/calculatedcontour_bins_scale1.npy'))
+        expectedSaliences1 = load(join(filedir(), 'pitchsalience/calculatedcontour_saliences_scale1.npy'))
+        expectedStartTimes = load(join(filedir(), 'pitchsalience/calculatedcontour_startTimes_scale.npy'))
+   
+        expectedBinsList0 = expectedBins0.tolist()
+        expectedSaliencesList0 = expectedSaliences0.tolist()
+        expectedBinsList1 = expectedBins1.tolist()
+        expectedSaliencesList1 = expectedSaliences1.tolist()
+        expectedStartTimesList = expectedStartTimes.tolist()
+        expectedDuration = len(audio)/(44100)
+
+        self.assertAlmostEqualVectorFixedPrecision(expectedBinsList0, contours_bins[0], 8)
+        self.assertAlmostEqualVectorFixedPrecision(expectedSaliencesList0, contours_saliences[0], 8)
+        self.assertAlmostEqualVectorFixedPrecision(expectedBinsList1, contours_bins[1], 8)
+        self.assertAlmostEqualVectorFixedPrecision(expectedSaliencesList1, contours_saliences[1], 8)
+        self.assertAlmostEqualVectorFixedPrecision(expectedStartTimesList, contours_start_times, 8)                        
+        self.assertAlmostEqual( expectedDuration, duration, 8)                           
```

Comment:
We need to test whether the correct pitch contour is output analytically, that is without referencing the output of the algorithm itself. We know how the ideal output should look like.

Reply:
To do it analytically I need to add  run_pitch_contours_melody to the algorithm chain to complete end to end test and get the rounded original pitch values from the scales ladder. I can use the original scale values as the reference